chain-signatures/node/src/protocol/signature.rs

```diff
@@ -87,42 +89,67 @@ impl SignQueue {
             }
         } {
             let mut rng = StdRng::from_seed(request.entropy);
-            let subset = stable.keys().choose_multiple(&mut rng, threshold);
-            let proposer = **subset.choose(&mut rng).unwrap();
-            if subset.contains(&&self.me) {
-                let is_mine = proposer == self.me;
+            let subset = stable.keys().cloned().choose_multiple(&mut rng, threshold);
+            let in_subset = subset.contains(&self.me);
+            let proposer = *subset.choose(&mut rng).unwrap();
+            let is_mine = proposer == self.me;
+            let request_id = CryptoHash(request.request_id);
+
+            tracing::info!(
+                ?request_id,
+                ?subset,
+                ?proposer,
+                in_subset,
+                is_mine,
+                "sign queue: organizing request"
+            );
+
+            let sign_id = SignRequestIdentifier::new(
```

Comment:
It is a SignRequest that has request_id inside. Should we unify that? Not a blocker for this PR, but check my comment in https://github.com/sig-net/mpc/issues/183

Reply:
yeah this is a stopgap right now before I do a whole entire refactor of our sign queue and signature id